src/fromager/wheels.py

```diff
@@ -129,6 +131,73 @@ def analyze_wheel_elfdeps(
     return requires, provides
 
 
+def add_extra_metadata_to_wheels(
+    ctx: context.WorkContext,
+    req: Requirement,
+    version: Version,
+    extra_environ: dict[str, str],
+    sdist_root_dir: pathlib.Path,
+    wheel_file: pathlib.Path,
+) -> None:
+    extra_data_plugin = overrides.find_override_method(
+        req.name, "add_extra_metadata_to_wheels"
+    )
+    data_to_add = {}
+    if extra_data_plugin:
+        data_to_add = overrides.invoke(
+            extra_data_plugin,
+            ctx=ctx,
+            req=req,
+            version=version,
+            extra_environ=extra_environ,
+            sdist_root_dir=sdist_root_dir,
+        )
+        if not isinstance(data_to_add, dict):
+            logger.warning(
+                f"{req.name}: unexpected return type from plugin add_extra_metadata_to_wheels. Expected dictionary. Will ignore"
+            )
+            data_to_add = {}
+
+    with tempfile.TemporaryDirectory() as dir_name:
```

Comment:
Have a look at #104, it shows how to use a class in auditwheel to edit wheel contents and automatically repack the file.

Reply:
It doesn't seem like auditwheel was made to be used as an API? It also has a non python dependency: https://github.com/pypa/auditwheel?tab=readme-ov-file#requirements

I would prefer to stick to `wheel` for now until the wheel API PR lands https://github.com/pypa/packaging/pull/805